storage/src/main/java/tech/pegasys/teku/storage/api/Eth1DepositChannel.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.storage.api;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+import tech.pegasys.teku.pow.event.DepositsFromBlockEvent;
+import tech.pegasys.teku.pow.event.MinGenesisTimeBlockEvent;
+import tech.pegasys.teku.util.async.SafeFuture;
+
+public interface Eth1DepositChannel {
+
+  SafeFuture<Stream<DepositsFromBlockEvent>> streamDepositFromBlockEvents();
```

Comment:
We shouldn't be returning a `Stream` here.  This should just be `SafeFuture<WherePowServiceShouldStartFrom> replayDepositEvents()`.  The returned value will actually be read from a different thread so you'd wind up accessing the `Stream` from multiple threads and its all going to get weird.

Generally anything that goes via a channel should still work if it were serialised and sent to the receiver running in another process (so it winds up with a different instance).  In some cases, we'll actually do that in the future but it's a useful model to follow anyway to make the boundaries between components clearer.


The `WherePowServiceShouldStartFrom` is probably just the block number that the last deposit came from and whether the min genesis event has already fired or not.  The PoWService would then ignore all the blocks equal to or before that block number (ie start it's subscription to deposit log events from that block number + 1).